api/app/routes/admin.py

```
@@ -250,3 +255,43 @@ async def get_metrics():
+
+
+@router.post("/faqs", response_model=FAQIdentifiedItem, status_code=201)
+async def add_new_faq_route(faq_item: FAQItem):
+    """Add a new FAQ."""
+    logger.info(f"Admin request to add new FAQ: {faq_item.question[:30]}...")
+    try:
+        new_faq = faq_service.add_faq(faq_item)
+        return new_faq
+    except Exception as e:
+        logger.error(f"Failed to add FAQ: {e}", exc_info=True)
+        raise HTTPException(status_code=500, detail="Failed to add FAQ.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)